app/assets/javascripts/form-validator.js

```diff
@@ -37,15 +37,23 @@ var FormValidator = function(root) {
 
   this.addInputValidation = function(inputId, validator) {
     var onChange = function (input) {
+      var inputValidationError = root.getElementById(input.id + 'ValidationError');
       if (validator.isValid(input.value)) {
+        inputValidationError.classList.add('display-none');
         input.classList.add('valid');
       } else {
+        if(input.value.length>0){
+          inputValidationError.classList.remove('display-none');
+        }
         input.classList.remove('valid');
+
       }
     };
     var input = document.getElementById(inputId);
     input.addEventListener('change', onChange.bind(this, input));
```

Comment:
Probably for both as it is used. ;) What I do not like about target and so is, that you have to care about browser compatibilities. And so it is much clearer I find :)